app/Jobs/ImportProcessJob.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+namespace App\Jobs;
+
+use App\Events\ImportProcessComplete;
+use App\Models\ImportProcess;
+use App\Models\Transaction;
+use Illuminate\Bus\Queueable;
+use Illuminate\Contracts\Queue\ShouldQueue;
+use Illuminate\Foundation\Bus\Dispatchable;
+use Illuminate\Queue\InteractsWithQueue;
+use Illuminate\Queue\SerializesModels;
+use PhpOffice\PhpSpreadsheet\Reader\Csv;
+
+class ImportProcessJob implements ShouldQueue
+{
+    use Dispatchable, InteractsWithQueue, Queueable, SerializesModels;
+
+    /** @var ImportProcess $fileImport */
+    public $fileImport;
+
+    /**
+     * Create a new job instance.
+     *
+     * @return void
+     */
+    public function __construct($fileImport)
+    {
+        $this->fileImport = $fileImport;
+    }
+
+    /**
+     * Execute the job.
+     *
+     * @return void
+     */
+    public function handle()
+    {
+        if (!$this->fileImport->imported) {
+            $file = $this->fileImport->fileName;
+            $reader = new Csv();
+            $spreadsheet = $reader->load(public_path().'/'.$file);
+            $worksheet = $spreadsheet->getActiveSheet();
+            $data = $worksheet->toArray();
+            $isHeader = true; // first row is header
+            foreach ($data as $row) {
+                if ($isHeader) {
+                    $isHeader = false;
+                    continue;
+                } else {
+                    $transaction = new Transaction();
+                    $transaction->user_id = $this->fileImport->user_id;
+                    $transaction->label = $row[0];
+                    $transaction->amount = floatval($row[1]);
+                    $transaction->date = $row[2];
+                    $transaction->save();
```

Comment:
This won't fulfill this requirement:

> Imported entries should not appear in the balance list, until the entire import has completed.

Reply:
Temporarily set user_id to null and updated it when import completes.